### PR TITLE
Log about configured but unavailable package managers 

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: utils/common/src/main/kotlin/Extensions.kt [hunk 1]
```diff
@@ -33,6 +33,11 @@ import java.nio.file.attribute.BasicFileAttributes
 import java.util.EnumSet
 import java.util.Locale
 
+/**
+ * Conveniently call [also] only if the receiver is null, e.g. for error handling, and return the receiver in any case.
```
Comment (fviernau) [inline utils/common/src/main/kotlin/Extensions.kt:37]: nit: "Conveniently" could be dropped.

File: utils/common/src/main/kotlin/Extensions.kt [hunk 2]
```diff
@@ -33,6 +33,11 @@ import java.nio.file.attribute.BasicFileAttributes
...
+ */
+inline fun <T> T.alsoIfNull(block: (T) -> Unit): T = this ?: also(block)
```
Comment (fviernau) [inline utils/common/src/main/kotlin/Extensions.kt:39]: Should / can this have the `T?` ? `inline fun <T> T?.alsoIfNull`

Comment (fviernau) [reply utils/common/src/main/kotlin/Extensions.kt:39]: The code block is only executed if `T` is null, so the parameter could be
`() -> Unit` instead.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: utils/common/src/main/kotlin/Extensions.kt (hunk 2 above)
Comment (sschuberth) [reply utils/common/src/main/kotlin/Extensions.kt:39]: > Should / can this have the `T?` ? `inline fun <T> T?.alsoIfNull`

No, in contrast to `Any.alsoIfNull` (which does not match `Any?`), using a generic `T` includes `T?`. FYI, I copied the function signature from Kotlin's `also` extension function.

Comment (sschuberth) [reply utils/common/src/main/kotlin/Extensions.kt:39]: > The code block is only executed if `T` is null, so the parameter could be `() -> Unit` instead.

No, it couldn't, as I'm passing `block` on to `also`, which requires this function signature.

Comment (fviernau) [reply utils/common/src/main/kotlin/Extensions.kt:39]: > No, it couldn't, as I'm passing block on to also

Hm, I basically meant something like the following:
```
inline fun <T> T.alsoIfNull(block: () -> Unit): T = this ?: run {
    block()
    null
}
```
e.g. not calling `also` at all. What do you think ?

Comment (sschuberth) [reply utils/common/src/main/kotlin/Extensions.kt:39]: I prefer calling `also` internally as it's slighly more compact and highlights the relation between these two functions.

**Review by fviernau — APPROVED** [review]

**Pull request merged**